pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -798,8 +780,6 @@ func generateMigrationParams(dom cli.VirDomain, vmi *v1.VirtualMachineInstance,
 	params := &libvirt.DomainMigrateParameters{
 		Bandwidth:    bandwidth, // MiB/s
 		BandwidthSet: bandwidth > 0,
-		URI:          migrURI,
-		URISet:       true,
```

Comment:
any idea why we don't need to set the migration URI anymore in these parameters? 

I'm curious how the block migration works now that we're using a unix socket for the libvirt connection. I think that connection still occurs over tcp proxied over local host

Reply:
That would be good to know. According to the bugzilla, at least on the cli there is now an argument for the disk migration socket: `virsh migrate machinename --verbose --compressed --live --copy-storage-all 'qemu+unix:///system?socket=/tmp/test-sock-driver' unix://?socket=/tmp/test-sock-qemu --disks-socket /tmp/test-sock-nbd`. That means that there should be something in the c-bindings too.

Reply:
> any idea why we don't need to set the migration URI anymore in these parameters?

Quoting from https://libvirt.org/html/libvirt-libvirt-domain.html#VIR_MIGRATE_PARAM_URI

`It takes a hypervisor specific format. The uri_transports element of the hypervisor capabilities XML includes details of the supported URI schemes. When omitted libvirt will auto-generate suitable default URI. It is typically only necessary to specify this URI if the destination host has multiple interfaces and a specific interface is required to transmit migration data.`

That said I don't think we need this at all since libvirt will autogenerate it and we migrate over a unix socket.
> 
> I'm curious how the block migration works now that we're using a unix socket for the libvirt connection. I think that connection still occurs over tcp proxied over local host

That still works over the TCP proxy, I actually did not change that because I did not understand how those types of migrations get started.. maybe you can enlighten me offline ? :)

Reply:
> That would be good to know. According to the bugzilla, at least on the cli there is now an argument for the disk migration socket: `virsh migrate machinename --verbose --compressed --live --copy-storage-all 'qemu+unix:///system?socket=/tmp/test-sock-driver' unix://?socket=/tmp/test-sock-qemu --disks-socket /tmp/test-sock-nbd`. That means that there should be something in the c-bindings too.

https://libvirt.org/html/libvirt-libvirt-domain.html#VIR_MIGRATE_PARAM_DISKS_URI

Reply:
> That still works over the TCP proxy, I actually did not change that because I did not understand how those types of migrations get started.. maybe you can enlighten me offline ? :)

I think you have to set `DisksURISet` and  `DisksURI` to the socket in `live-migration-source.go`. That should make all other tcp sockets obsolete.

Reply:
> > That still works over the TCP proxy, I actually did not change that because I did not understand how those types of migrations get started.. maybe you can enlighten me offline ? :)
> 
> I think you have to set `DisksURISet` and `DisksURI` to the socket in `live-migration-source.go`. That should make all other tcp sockets obsolete.

Sorry my fault, that part is clear, I don't understand what happens on the target pod. 
Specifically I don't understand where's the link with libvirt, from what I could see on the target virt-launcher we open these 2 tcp proxies:

/var/run/kubevirt/migration-proxy/VMI_UID-49152-source.sock <-> 127.0.0.1:49152
/var/run/kubevirt/migration-proxy/VMI_UID-49153-source.sock <-> 127.0.0.1:49153

I then cannot find where does the data get forwarded once it reaches these 2 ports on localhost, is libvirt listening on those 2 ports?

The whole path from the source pod to target pod is clear, it's the last step that I'm missing (how the migration data actually reaches the target libvirt), and this is the reason why I did not touch those 2 proxies.

Reply:
> /var/run/kubevirt/migration-proxy/VMI_UID-49152-source.sock <-> 127.0.0.1:49152

That is the start of the default port range which libvirt will use for migrations: https://gitlab.com/libvirt/libvirt/-/blob/master/src/qemu/qemu_conf.c#L71

Reply:
ah right! now it makes sense! thank you :+1:

Reply:
@acardace do you plan to remove the storage proxies in a follow-up?

Reply:
You asked while I was typing I guess :) https://github.com/kubevirt/kubevirt/pull/6070#issuecomment-887347928

Reply:
Yep :)